api/lib/opentelemetry/distributed_context/propagation/http_text_format.rb

```diff
@@ -17,7 +16,35 @@ module Propagation
       # Propagation is usually implemented via library-specific request interceptors, where the client-side injects values
       # and the server-side extracts them.
       class HTTPTextFormat
-        # TODO
+        # extract will return a SpanContext from the supplied carrier
+        # invalid headers will result in a new SpanContext
+        # @param [Carrier] the carrier to get the header from
+        # @yield [Carrier, String] the header key
+        # {SpanContext}
+        def extract(carrier)
+          raise Error, 'block must be supplied' unless block_given?
+
+          header = yield carrier, TraceParent::TRACE_PARENT_HEADER
+          tp = TraceParent.from_string(header)
+
+          SpanContext.new(trace_id: tp.trace_id, span_id: tp.span_id, flags: tp.flags)
+        rescue OpenTelemetry::Error
+          SpanContext.new
+        end
+
+        # inject will set the span context on the supplied carrier
+        # @param [Context] the carrier
+        # @yield [Carrier, String, String] carrier, header key, header value
+        # {SpanContext}
```

Comment:
Don't need this. We don't return the `SpanContext` (at least not intentionally - maybe we should explicitly return `self`).